apps/spruce/src/components/Spawn/utils/hostUptime.ts

```diff
@@ -161,7 +161,17 @@ const getDailyUptime = ({
 }: {
   startTime: string;
   stopTime: string;
-}) => differenceInHours(new Date(stopTime), new Date(startTime));
+}) => {
+  const startDate = new Date(startTime);
+  const stopDate = new Date(stopTime);
+
+  // If this is an overnight schedule, set stop date to the following day so that uptime is correctly calculated.
+  if (stopDate < startDate) {
```

Comment:
How does this condition indicate an overnight schedule?

Reply:
If the stopTime occurs before the startTime, the back end interprets the start as occurring on the following day